[addons] code cleanups and improvement inside addon base classes

AlwinEsch opened the pull request:
## Description

This request is in the background with https://github.com/xbmc/xbmc/pull/20860 to make it smaller and clean up or improve some small things in advance.

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

## Motivation and context

To finally be able to do something with Kodi again and to be able to continue upcoming and active projects.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?

Will check the request as best I can on all add-on types, check it out so that it doesn't require any changes.

<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons] remove "std::string message" value from CAddonStatusHandler

The value "std::string& message" becomes not used, this cleans up by remove about.
Was come to it by note on another request.

AlwinEsch committed:
[addons] clang style cleanup on Addon.[cpp|h] and IAddon.h

Relates to have following changes more clean and prevent clang problems
as in IAddon.h was the two spaces on begin.

AlwinEsch committed:
[addons] use internal functions within AddonSettings.cpp on private namespace

Seen on cleanup thats they defined without a namespace, to have this more correct becomes they inserted into an private namespace on file.

AlwinEsch committed:
[addons] rename value instanceID to uniqueWorkID in IAddonInstanceHandler

In relationship on other request where a special instance identfier as integer becomes added, becomes this here renamed.
The uniqueWorkID is mainly thought to give from an add-on instance an identfier to be used on another add-on instance, e.g. PVR stream to use inputstream within his add-on.

AlwinEsch committed:
[addons] set profile path within CAddonInfo

This reduce some not needed string formats about same names and stores it now one time inside CAddonInfo and to get with ProfilePath() function.

AlwinEsch committed:
[addons] don't use CAddonPtr as return in CAddonSettings

On the one hand it is currently only called at one point in CAddonSettings to get the addon id (which is already stored in the class as a string) and on the other hand I don't like giving out the shared pointer of an add-on freely (if not absolutely necessary), because one it is never certain where it will be lurking around (in the case of an add-on update, for example).

AlwinEsch committed:
[addons] class construction cleanup on CAddon

Makes it more clean to match newest versions and to remove not needed sets.

AlwinEsch committed:
[addons] replace all "C" NULL to "C++" nullptr

Another cleanup about code style.

AlwinEsch committed:
[addons] clang style cleanup on AddonSettings.[cpp|h]

To cleanup and prevent unrelated changes on another request.

AlwinEsch commented:
> I've made some formatting changes to meet the current code style. The diffs are available in the following links:
> 
> * [PR21739-0003-6ae59dbe2d-addons-clang-style-cleanup-on-Addo.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/23515/artifact/PR21739-0003-6ae59dbe2d-addons-clang-style-cleanup-on-Addo.diff)
> * [PR21739.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/23515/artifact/PR21739.diff)
> 
> For more information please see our [current code style guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

No idea why he bring again his suggested change is in :thinking:.

ksooo reviewed (approved):
Looks good to me.

garbear reviewed (approved):
Can you add documentation about what "unique work ID" means in the code?

howie-f commented on xbmc/addons/settings/AddonSettings.cpp:
```diff
@@ -39,6 +39,9 @@
 #include <mutex>
 #include <vector>
 
+namespace
+{
+
 static const std::string OldSettingValuesSeparator = "|";
 
 static const int UnknownSettingLabelIdStart = 100000;
```
not sure if an anonymous namespace + static makes sense here. so i would change this to
```cpp
namespace
{
constexpr auto OldSettingValuesSeparator = "|";
constexpr int UnknownSettingLabelIdStart = 100000;
```

AlwinEsch replied:
Have changed to constexpr.

howie-f reviewed (approved):
nice cleanup with just one small suggestion

AlwinEsch committed:
[addons] CLang cleanup on xbmc/addons/AddonEvents.h

To have by following changes more correct and prevent additional line breaks.

AlwinEsch committed:
[addons] rename value id inside AddonEvent to addonId

The background here is to have it cleaner in the name, since "id" alone can be ambiguous. The other background is, in the case of subsequent changes where instance settings are added, to set the value as instanceId and more appropriately to addonId.

AlwinEsch committed:
[addons] use constexpr instead of static in anonymous namespace on AddonSettings.cpp

Static seems no more makes sense in an anonymous namespace.

AlwinEsch commented:
> unique work ID

Have added on one commit, looks maybe a bit big, but was not come with my bad english to something better and shorter.

Also added two another new commits where was come on rebase by https://github.com/xbmc/xbmc/pull/20860 today.

AlwinEsch commented:
Jenkins build this please

AlwinEsch commented on xbmc/addons/binary-addons/AddonInstanceHandler.h:
```diff
@@ -22,14 +22,36 @@ namespace ADDON
+   *                         since it is not known at the start what is being
+   *                         executed ( live TV, radio, recordings...) and add-on
+   *                         may use different classes.
+   */
```
@garbear can you look about the text, not sure how it can be better with my bad english :sweat_smile:.

AlwinEsch replied:
Have played a bit (not uploaded now):
```cpp
  /**
   * @brief Class constructor for handling add-on instance processes, allowing
   * an add-on to handle multiple work simultaneously and independently.
   *
   * @param[in] type The associated add-on type which is processed in the running
   *                 instance.
   * @param[in] addonInfo Class for querying available add-on information (e.g.
   *                      content declared in addon.xml).
   * @param[in] parentInstance [opt] Above running add-on instance which starts this
   *                           instance. Used to have the associated class for
   *                           work to open in the add-on.
   *                           Currently used places:
   * | Parent | Target | Description
   * |--------|--------|-------------
   * | @ref kodi::addon::CInstanceInputStream | @ref kodi::addon::CInstanceVideoCodec | In order to be able to access the overlying input stream instance in the video codec created by Kodi on the add-on.
   * @param[in] uniqueWorkID [opt] Identification value intended to pass any special
   *                         values to the instance to be opened.
   *                         If not used, the IAddonInstanceHandler class pointer
   *                         is used as a string.
   *                         Currently used values:
   * | Add-on instance type | Description
   * |----------------------|---------------------
   * | @ref kodi::addon::CInstanceInputStream "Inputstream" | To transfer special values to inputstream using the property @ref STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID from external space, for example PVR add-on which also supports inputstream can exchange special values with it, e.g. select the necessary add-on processing class, since it is not known at the start what is being executed ( live TV, radio, recordings...) and add-on may use different classes.
   * | All other | The used class pointer of Kodi's @ref IAddonInstanceHandler is used as a value to have an individually different value.
   */
  IAddonInstanceHandler(ADDON_TYPE type,
                        const AddonInfoPtr& addonInfo,
                        KODI_HANDLE parentInstance = nullptr,
                        const std::string& uniqueWorkID = "");
```

Screenshot:
![Screenshot_20220813_111202](https://user-images.githubusercontent.com/6879739/184477311-ec6f3bef-3eb3-419e-9c7a-32c28dfc963a.png)

AlwinEsch committed:
[addons] add class constructor docs to IAddonInstanceHandler

To have more infos about his values.

AlwinEsch commented:
Is rebased, if nothing against I bring in later to cleanup the other request.

howie-f reviewed (approved):
tested for a longer time, no issues obvious

ksooo commented:
> Is rebased, if nothing against I bring in later to cleanup the other request.

@AlwinEsch you have three approvals, that is more than enough. Go ahead and merge (once Jenkins is happy).
